mrmustard/math/backend_manager.py

```diff
@@ -1023,6 +1053,7 @@ def prod(self, array: Tensor, axis=None) -> Tensor:
         Returns:
             The product of the elements in ``array``.
         """
+        array = self.astensor(array)
```

Comment:
Like for `gather` I don't know if it's a good idea to call `astensor` automatically each time we call this function rather than outside, only as needed.

Reply:
if it's the same jax issue, no problem